src/KeyVault/KeyVault/Commands/Key/GetAzKeyVaultRandomNumber.cs

```diff
@@ -0,0 +1,97 @@
+﻿using Microsoft.Azure.Commands.KeyVault.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Text;
+
+namespace Microsoft.Azure.Commands.KeyVault.Commands.Key
+{
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzurePrefix + "KeyVaultRandomNumber", DefaultParameterSetName = GetByHsmNameParameterSet)]
+    [OutputType(typeof(string))]
+    public class GetAzKeyVaultRandomNumber: KeyVaultCmdletBase
+    {
+        #region Parameter Set Names
+        
+        private const string GetByHsmNameParameterSet = "GetByHsmName";
+        private const string GetByHsmInputObjectNameParameterSet = "GetByHsmInputObject";
+        private const string GetByHsmResourceIdParameterSet = "GetByHsmResourceId";
+
+        #endregion
+
+        #region Input Parameter Definitions
+
+        /// <summary>
+        /// HSM Name
+        /// </summary>
+        [Parameter(Mandatory = true,
+            Position = 0,
+            ParameterSetName = GetByHsmNameParameterSet,
+            HelpMessage = "HSM name. Cmdlet constructs the FQDN of a managed HSM based on the name and currently selected environment.")]
+        [ResourceNameCompleter("Microsoft.KeyVault/managedHSMs", "FakeResourceGroupName")]
+        [ValidateNotNullOrEmpty]
+        [Alias("HsmName")]
+        public string Name;
```

Comment:
Unless we are certain that this feature won't be available on normal vaults, I suggest we name this parameter `HsmName` and do not add any alias, because it's a convention that "Name" represents name of the vault.